### ISSUE TITLE
Inconsistent behavior of the extension's Capture Group Processing when using Replace Occurrences 

### ISSUE THREAD

**Issue by gkarag** [mshibanami/redirect-web#40]
Hello,

I am trying to remove particular URL parameters from the Youtube video URL's and I stumbled on inconsistent behavior from the extension. I am using the latest available version 6.7.0. Please find below the minimal examples demonstrating the inconsistency. 

- Redirect from (using Wildcard): `https://www.youtube.com/watch*`
- Redirect to: `$0`
- Capturing Group Processing, working example
-- Group: `$0`
-- Process: Replace Occurrences (using Regular Expression)
-- Target: `e`
-- Replacement: `y`
- Capturing Group Processing, nonworking example (results in a warning "This may cause an infinite loop" for the example URL below)
-- Group: `$0`
-- Process: Replace Occurrences (using Regular Expression)
-- Target: `r`
-- Replacement: `y`
- Example URL:  `https://www.youtube.com/watch?v=VVVVVVVVVV&embeds_referring_euri=xxxxxxxxxxxx&source_ve_path=yyyyyyyyy&feature=zzzzzzzzzzzzz`

**Comment by mshibanami** [comment]
Hi @gkarag, sorry for any inconvenience, and thanks for your feedback!

I've confirmed the issue and realized I need to improve the infinite loop detection. In the meantime, there is a workaround.

## Problem of the infinite loop detection

Regarding the first case (`e` → `y`), the result of the example rule modifies the URL from `youtube.com` to `www.youtuby.com`, which doesn't match `https://www.youtube.com/watch*`:

```
https://www.youtuby.com/watch?v=VVVVVVVVVV&ymbyds_ryfyrring_yuri=xxxxxxxxxxxx&sourcy_vy_path=yyyyyyyyy&fyatury=zzzzzzzzzzzzz
```

However, in the case of the infinite loop (`r` → `y`), the initial part of the URL `https://www.youtube.com/watch` is not modified:

```
https://www.youtube.com/watch?v=VVVVVVVVVV&embeds_yefeyying_euyi=xxxxxxxxxxxx&souyce_ve_path=yyyyyyyyy&featuye=zzzzzzzzzzzzz
```

As a result, this still matches `https://www.youtube.com/watch*`, and Redirect Web considers it may cause an infinite loop.

However, this actually doesn't cause an infinite loop because if you access the above URL, Redirect Web skips it as  the same URL. In other words, the current Redirect Web doesn't check whether the loop will end by skipping the same URL.

I'll improve the implementation to address this logic. Thanks!

### Workaround

You can include the unneeded parameter into Redirect To so that the URL without a parameter won't be targeted by Redirect To.
Specifically, suppose you want to remove the `embeds_referring_euri` parameter, change Redirect To to this: 

```
https://www.youtube.com/watch\?.*&embeds_referring_euri=.*
(Regular Expression)
```

The  [Getting Started](https://mshibanami.github.io/redirect-web/#/getting-started for more details) may also help you.
But it's just a workaround. As I said, I'll fix the logic soon.

**Comment by gkarag** [comment]
Hello, @mshibanami and thank you for the prompt response and the suggestions :) 

For the record I was trying a regular expression in the Capture Group Processing that would remove any chosen parameter, which I would consider irrelevant, i.e. replace target matching a regular expression like`[&]*(embeds_referring_euri|source_ve_path|feature)=[^&]*` with an empty string. After I unsuccessfully tried different approaches,I have reduced the example to the simplest possible to demonstrate the problem.

Thanks once more and kind regards.

**Comment by mshibanami** [comment]
@gkarag I've released a new version, 6.8.0, which should fix the issue.
Although you already solved your issue, I hope it'll help you in the future. Thank YOU!

I'll close this issue now, but please feel free to reopen it. 🙂

**Comment by gkarag** [comment]
The new fixed version worked for what I was trying to do. Many thanks 🙏🏻
